File: back-end/src/main/java/com/trucks/application/service/VesselService.java
package com.trucks.application.service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;

import com.trucks.application.port.VesselRepository;
import com.trucks.domain.entity.Vessel;

import org.springframework.web.server.ResponseStatusException;
import org.springframework.http.HttpStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public class VesselService {

    @Autowired
    VesselRepository vesselRepository;

    public Page<Vessel> findAll(PageRequest pageRequest) {
        return this.vesselRepository.findAll(pageRequest);
    }

    public Vessel save(Vessel newVessel) {
        return this.vesselRepository.save(newVessel);
    }

    public Optional<Vessel> find(UUID id) {
        return this.vesselRepository.find(id);
    }

    public Vessel delete(UUID id) {
        return this.vesselRepository.delete(id);
    }

    public Vessel update(UUID id, Vessel updatedVessel) {
        return this.vesselRepository.update(id, updatedVessel);
    }
    

    public List<Vessel> deleteAll() {
        return this.vesselRepository.deleteAll();
    }
}

File: front-end/src/application/service/vesselService.ts
import { VesselRepository } from "../repository/vesselRepository";
import { Vessel } from "../../domain/entity/vessel";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from "../../utils/ull-uuid";

export class VesselService {
    private vesselRepository: VesselRepository;

    constructor(vesselRepository: VesselRepository) {
        this.vesselRepository = vesselRepository;
    }
    
    public save = (newVessel: Vessel): Promise<Either<DataError, Vessel>> => {
        return this.vesselRepository.save(newVessel);
    }
    
    public find = (id: UllUUID): Promise<Either<DataError, Vessel>> => {
        return this.vesselRepository.find(id);
    }
    
    public findAll = (pageRequest: any): Promise<Either<DataError, Vessel[]>> => {
        return this.vesselRepository.findAll(pageRequest);
    }

    public update = (id: UllUUID, updatedVessel: Vessel): Promise<
        Either<DataError, Vessel>> => {
            return this.vesselRepository.update(id, updatedVessel);
    }
    
    public delete = (id: UllUUID): Promise<Either<DataError, Vessel>> => {
        return this.vesselRepository.delete(id);
    }

    public deleteAll = (): Promise<Either<DataError, Vessel[]>> => {
        return this.vesselRepository.deleteAll();
    }
}